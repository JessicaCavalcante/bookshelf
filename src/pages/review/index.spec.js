import { Review } from './index.js';
import { createReview } from '../../services/index.js';

describe('Review', () => {
  it('should be a function', () => {
    expect(typeof Review).toBe('function');
  });

  it('loads the page', () => {
    expect(Review()).toMatchSnapshot();
  });

  it('clicks to send review', () => {
    const reviewPage = Review();
    reviewPage.querySelector('#btn-review').dispatchEvent(new Event('click'));
    expect(createReview).toHaveBeenCalled();
  });
});
