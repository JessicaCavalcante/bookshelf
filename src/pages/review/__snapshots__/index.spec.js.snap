// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Review should load the page 1`] = `
<div
  class="container"
>
  
      
  <header>
    
        
    <h2
      class="title-review"
    >
      Publicar nova resenha
    </h2>
    
      
  </header>
  
      
  <section>
    
        
    <form
      class="form-review"
    >
      
        
      <label
        class="label-review"
      >
        Livro: 
        <span
          class="required"
        >
          *
        </span>
      </label>
      
        
      <input
        class="input-book"
        id="book-name"
        name="book"
        required=""
        type="text"
      />
      
        
      <label
        class="label-review"
      >
        Autor: 
        <span
          class="required"
        >
          *
        </span>
      </label>
      
        
      <input
        class="input-book"
        id="book-author"
        name="author"
        required=""
        type="text"
      />
      
        
      <label
        class="label-review"
      >
        Resenha: 
        <span
          class="required"
        >
          *
        </span>
      </label>
      
        
      <textarea
        class="review"
        id="book-review"
        required=""
      />
      
        
      <button
        class="btn-review"
        id="btn-review"
        type="submit"
      >
        Publicar
      </button>
      
        
    </form>
    
      
    <section>
      
  
    </section>
  </section>
</div>
`;
